File: csharp/Platform.Data.Doublets.Gql.Schema/OrderBy.cs

namespace Platform.Data.Doublets.Gql.Schema
{
    public enum order_by
    {
        asc,
        asc_nulls_first,
        asc_nulls_last,
        desc,
        desc_nulls_first,
        desc_nulls_lasr
    }

    class OrderBy
    {
        public OrderBy from { get; set; }

        public order_by? from_id { get; set; }

        public order_by? id { get; set; }

        public OrderBy to { get; set; }

        public order_by? to_id { get; set; }

        public OrderBy type { get; set; }

        public order_by? type_id { get; set; }
    }
}

File: rust/src/model/enum_type/order_by.rs
use crate::model::LinkType;
use async_graphql::*;
use std::cmp::Ordering;

#[derive(Enum, Debug, Copy, Clone, Eq, PartialEq)]
#[graphql(name = "order_by")]
pub enum OrderBy {
    #[graphql(name = "asc")]
    Asc,
    #[graphql(name = "asc_nulls_first")]
    AscNullsFirst,
    #[graphql(name = "asc_nulls_last")]
    AscNullsLast,
    #[graphql(name = "desc")]
    Desc,
    #[graphql(name = "desc_nulls_first")]
    DescNullsFirst,
    #[graphql(name = "desc_nulls_last")]
    DescNullsLast,
}

impl OrderBy {
    pub fn matches(&self, a: LinkType, b: LinkType) -> Ordering {
        match self {
            OrderBy::Asc => a.cmp(&b),
            OrderBy::AscNullsFirst => a.cmp(&b),
            OrderBy::AscNullsLast => b.cmp(&b),
            OrderBy::Desc => b.cmp(&a),
            OrderBy::DescNullsFirst => b.cmp(&a),
            OrderBy::DescNullsLast => b.cmp(&a),
        }
    }
}
